postreise/analyze/generation/capacity.py

```diff
@@ -109,3 +112,26 @@ def sum_capacity_by_type_zone(scenario):
     plant = grid.plant
 
     return plant.groupby(["type", "zone_id"])["Pmax"].sum().unstack().fillna(0)
+
+
+def get_capacity_factor_time_series(scenario, area, resources, area_type=None):
+    """Get time series capacity factor for every plant of the resources in the
+    specific area.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance
+    :param str area: one of *loadzone*, *state*, *state abbreviation*,
+        *interconnect*, *'all'*
+    :param str/list resources: one or a list of resources
+    :param str area_type: one of *'loadzone'*, *'state'*, *'state_abbr'*,
+        *'interconnect'*
+    :return: (*pandas.DataFrame*) -- index: timestamps, column: plant ids,
+        value: capacity factors
+    """
+    _check_scenario_is_in_analyze_state(scenario)
+    plant_id = get_plant_id_for_resources_in_area(
+        scenario, area, resources, area_type=area_type
+    )
+    pg = scenario.state.get_pg()[plant_id]
```

Comment:
`scenario.state.get_pg()[plant_id]` --> `scenario.get_pg()[plant_id]`

Reply:
The reason I keep it in the old format is because our Mock object doesn't have the alias which will break all of the tests.

Reply:
Thanks for reminding me. Since it is internal, it is transparent to the user it does not matter how we access things

Reply:
Should we create a separate test for the pass-through? Something like `assert scenario.state.x is scenario.x` for several relevant `x`?

Reply:
@danielolsen Where would such tests locate? Should we put it somewhere in `PowerSimData` rather than `PostREISE` @rouille ?

Reply:
Definitely in PowerSimData. I just saw this conversation so I commented here.

Reply:
I would create some tests in PowerSimData to make sure that `scenario.state.x is scenario.x` as @danielolsen suggested and I would refactor the Mocks to make sure it can take both. This PR does not have to wait for the functionality though. Let's create an issue so we don't forget about it.

Reply:
Issues are created, [253](https://app.zenhub.com/workspaces/renewable-energy-integration-5bfdd1184b5806bc2bf88976/issues/breakthrough-energy/postreise/253) and [433](https://app.zenhub.com/workspaces/renewable-energy-integration-5bfdd1184b5806bc2bf88976/issues/breakthrough-energy/powersimdata/433).